exec/java-exec/src/main/java/org/apache/drill/exec/rpc/control/ConnectionManagerRegistry.java

```diff
@@ -32,24 +29,19 @@
 
   private final ConcurrentMap<DrillbitEndpoint, ControlConnectionManager> registry = Maps.newConcurrentMap();
 
-  private final ControlMessageHandler handler;
-  private final BootStrapContext context;
-  private volatile DrillbitEndpoint localEndpoint;
-  private final BufferAllocator allocator;
+  private final BitConnectionConfigImpl config;
 
-  public ConnectionManagerRegistry(BufferAllocator allocator, ControlMessageHandler handler, BootStrapContext context) {
-    super();
-    this.handler = handler;
-    this.context = context;
-    this.allocator = allocator;
+  public ConnectionManagerRegistry(BitConnectionConfigImpl config) {
```

Comment:
Each impl of ConnectionConfig is package private. There are subtle differences among them, which is why references using impl are used with within the package. But in generic classes (e.g.AbstractServerConnection), interface is used.